contrib/python-sdk/openpaisdk/cluster.py

```diff
@@ -0,0 +1,212 @@
+        return dic[alias]
+
+    def attach_storage(self, alias: str, storage: dict, as_default: bool=False):
+        cluster = self.select(alias)
+        result = Cluster.attach_storage(cluster, storage)
+        if as_default:
+            cluster["default_storage_alias"] = storage["storage_alias"]
+        return result
+
+    def get_client(self, alias: str):
+        return ClusterClient(**self.select(alias))
+
+
+class ClusterClient:
+    """A wrapper of cluster to access the REST APIs"""
+
+    def __init__(self, pai_uri: str=None, user: str=None, password: str=None, storages: list=[], default_storage_alias: str=None, **kwargs):
+        __logger__.debug('creating cluster from info %s', get_args())
+        self.pai_uri = pai_uri
+        self.user = user
+        self.password = password
+        self.storages = storages
+        self.default_storage_alias = default_storage_alias
+        self.storage_clients = {}
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+        for cfg in storages:
+            self.add_storage(**cfg)
+        if len(storages) == 1:
+            self.default_storage_alias = storages[0]["storage_alias"]
+
+    def get_storage(self, alias: str=None):
+        if len(self.storage_clients) == 0:
+            return None
+        return self.storage_clients[alias if alias else self.default_storage_alias]
+
+    def add_storage(self, protocol: str=None, storage_alias: str=None, **kwargs):
+        "initialize the connection information"
+        func = 'add_storage_%s' % protocol.lower()
+        return getattr(self, func)(storage_alias, **kwargs)
+
+    def add_storage_webhdfs(self, storage_alias, web_hdfs_uri: str, **kwargs):
+        self.storage_clients[storage_alias] = Storage(protocol='webHDFS', url=web_hdfs_uri, user=kwargs.get('user', self.user))
+        return self
+
+    def get_token(self, expiration=3600):
+        """
+        [summary]
+            expiration (int, optional): Defaults to 3600. [description]
+
+        Returns:
+            OpenPAIClient: self
+        """
+
+        self.token = self.rest_api_token(expiration)
+        return self
+
+    def get_job_link(self, job_name: str):
+        return '{}/job-detail.html?username={}&jobName={}'.format(self.pai_uri, self.user, job_name)
+
+    def jobs(self, job_name: str=None, name_only: bool=False):
+        """
+        query the list of jobs
+            jobName (str, optional): Defaults to None. [description]
+            name_only (bool, optional): Defaults to False. [description]
+
+        Returns:
+            [type]: [description]
+        """
+
+        job_list = self.rest_api_jobs(job_name)
+        return [j['name'] for j in job_list] if name_only else job_list
+
+    def rest_api_jobs(self, job_name: str=None, info: str=None, user: str=None):
+        user = self.user if user is None else user
+        pth = '{}/rest-server/api/v1/user/{}/jobs'.format(self.pai_uri, user)
```

Comment:
You should check if pylon is enabled before using link like `/rest-server`.
If pylon is disabled, user needs to specify the rest server's port/url itself.

Reply:
add dependencies description in the README as below

> Pylon is required to parse the REST api path like /reset-server/.